File: src/main/java/hu/robnn/reportserver/model/dto/Driver.kt
package hu.robnn.reportserver.model.dto

import hu.robnn.commons.interfaces.UuidHolder
import hu.robnn.reportserver.enums.DbType
import java.util.*

data class Driver(var uuid: UUID = UUID.randomUUID(),
                  var name: String? = null,
                  var driverClassName: String? = null,
                  var dbType: DbType? = null): UuidHolder {
    override fun setUuid(p0: String?) {
        if(!p0.isNullOrEmpty()) {
            uuid = UUID.fromString(p0!!)
        }
    }

    override fun getUuid(): String {
        return uuid.toString()
    }
}

data class Drivers(val drivers: List<Driver?> = mutableListOf())

File: reportserver-ui/src/app/model/driver.ts
export enum DbType {
    ORACLE_THIN = "ORACLE_THIN",
    MY_SQL = "MY_SQL",
    MICROSOFT_SQL_SERVER = "MICROSOFT_SQL_SERVER",
    POSTGRESQL = "POSTGRESQL",
    DB2 = "DB2",
    SQLITE = "SQLITE",
}

export class Driver {
    uuid: string;
    name: string;
    driverClassName: string;
    dbType: DbType;
}

export class Drivers {
    drivers: Driver[];
}